test/verifier_test.js

```diff
@@ -1979,4 +1985,27 @@ describe("Verifier tests", function() {
       a.apiMock.triggerEvent("WorkersParameterized", event);
     });
   });
+
+  it("Wrong output hash of failed task submission pre-mined", async function() {
+    const tree = TEST_TREE.verifier;
+    if (!tree["all"] || !tree["#48"]) {
+      this.skip();
+    }
+
+    return new Promise(async function(resolve) {
+      let a = await initStuffForTaskSubmission();
+      let task = new FailedResult(a.taskId, constants.TASK_STATUS.FAILED, a.output, 5, "signature");
+      let status = constants.ETHEREUM_TASK_STATUS.RECORD_CREATED;
+
+      a.apiMock.setTaskParams(a.taskId, a.blockNumber, status);
+      a.verifier.verifyTaskSubmission(task, a.blockNumber, a.taskId, null).then(res => {
+        assert.strictEqual(res.isVerified, false);
+        assert.strictEqual(res.error instanceof errors.TaskVerificationErr, true);
+        resolve();
```

Comment:
I think that if a test returns a Promise you don't need to resolve it.

Reply:
It is required